core-bundle/src/DependencyInjection/ContaoCoreExtension.php

```diff
@@ -509,4 +510,20 @@ private function getComposerPublicDir(string $projectDir): string|null
 
         return Path::join($projectDir, $publicDir);
     }
+
+    private function handleSecurityConfig(array $config, ContainerBuilder $container): void
+    {
+        if (!$container->hasDefinition('contao.listener.transport_security_header')) {
+            return;
+        }
+
+        if (false === $config['security']['hsts']['enabled']) {
+            $container->removeDefinition('contao.listener.transport_security_header');
```

Comment:
Should we also disable the header on the `RedirectController` if it is disabled in the config?

Reply:
I don't think that would be correct. Imho, you would always want that behavior.